### DISCUSSION TITLE
Not widening when expected

### DISCUSSION THREAD

**Discussion by phil-nelson** [renovatebot/renovate#10747] [Request Help]
I would like renovate to be widening by default for a PHP library I have. E.g.

`"psr/container": "~1.0",`

should be updated to

`"psr/container": "~1.0 || ~2.0",`

However, the PR it is creating is replacing with:

`"psr/container": "~2.0",`

My renovate.json looks like:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "composerIgnorePlatformReqs": false,
  "rangeStrategy": "widen",
  "packageRules": [
      {
          "matchDepTypes": ["require-dev"],
          "rangeStrategy": "bump"
      }
  ]
}
```

My `require-dev` dependencies look like they are getting bumped properly, however the PR renovate created for `psr/container` also looks like a bump (which is a `require`, not `require-dev`).

Can anyone tell me why it's doing this?

**Comment by phil-nelson** [comment]
Looking at https://github.com/renovatebot/renovate/blob/main/lib/versioning/composer/index.ts it seems that composer versioning does not support widening unless there is already a ` || `. I tried adding a test:

```
    expect(
      semver.getNewValue({
        currentValue: '^1.2',
        rangeStrategy: 'widen',
        currentVersion: '1.2.0',
        newVersion: '2.0.0',
      })
    ).toEqual('^1.2 || ^2.0');
```

which fails.

Is it supposed to work the way I am expecting?

**Reply by viceice** [reply]
You are right, `widen` is only supported when there is already a `||`. 

@rarkins Do you know something?

It seems that hare we miss a else?
https://github.com/renovatebot/renovate/blob/74d7691befa635c1220d8569936273bfb950e748/lib/versioning/composer/index.ts#L175-L175

Shouldn't this be lifted to outside the `||` check?
https://github.com/renovatebot/renovate/blob/74d7691befa635c1220d8569936273bfb950e748/lib/versioning/composer/index.ts#L185-L187

**Reply by phil-nelson** [reply]
@viceice @rarkins I have created https://github.com/renovatebot/renovate/pull/10753 , I admit I do not know a huge amount about renovate and composer versioning but this appears to fix this case, would appreciate some eyes on it. Thanks!
